src/code/actions.js

```diff
@@ -290,25 +297,55 @@ function _submitDrake(targetDrakeIndex, userDrakeIndex, correct, state) {
 
 export function submitDrake(targetDrakeIndex, userDrakeIndex, correct, incorrectAction) {
   return (dispatch, getState) => {
-    
+
 
     const state = getState();
     dispatch(_submitDrake(targetDrakeIndex, userDrakeIndex, correct, state));
 
+    const levels = state.authoring,
+          levelCount = levels.length,
+          missions = levels[state.routeSpec.level],
+          missionCount = missions.length,
+          challenges = missions[state.routeSpec.mission],
+          challengeCount = challenges.length,
+          trials = state.trials,
+          trialCount = trials.length;
     let challengeComplete = false,
-        missionComplete = false;
+        missionComplete = false,
+        // levelComplete = false,
```

Comment:
I left this one in because I anticipate that we will want to give a unique end-of-level message at some point.